proposals/0016-simd-application-fees.md

```diff
@@ -0,0 +1,278 @@
+---
+simd: '0016'
+title: Application Fees (Write-lock fees)
+authors:
+  - Godmode Galactus (Mango Markets)
+  - Maximilian Schneider (Mango Markets)
+category: Standard
+type: Core, Fees
+status: Draft
+created: 2022-12-23
+feature:
+---
+
+
+## Summary
+
+This SIMD will discuss additional fees called Application Fees or write lock fees.
+These fees are decided and set by the dapp developer to interact with the dapp. 
+Dapp developers then can decide to rebate these fees if a user interacts with the dapp 
+as intended and disincentivize the users which do not.
+These fees will be applied even if the transaction eventually fails and collected on 
+the same writable account.
+Account authority (i.e owner of the account) can do lamport transfers to recover these fees.
```

Comment:
This in theory sounds okay as long as the auth scheme is implemented in as general a manner as possible.

i.e. the auth can live on a spectrum from 100% trustless to 100% controlled where the methodology is transparent

Reply:
It's really on the dapp developer to decide this.